keel-titus-plugin/src/test/kotlin/com/netflix/spinnaker/keel/titus/resource/TitusClusterHandlerTests.kt

```diff
@@ -433,6 +440,52 @@ class TitusClusterHandlerTests : JUnit5Minutests {
             .containsDistinctElements()
         }
       }
+      context("the cluster has active server groups") {
+        before {
+          coEvery { cloudDriverService.titusActiveServerGroup("us-east-1") } returns activeServerGroupResponseEast
+          coEvery { cloudDriverService.titusActiveServerGroup("us-west-2") } returns activeServerGroupResponseWest
+        }
+
+        derivedContext<SubmittedResource<TitusClusterSpec>>("exported titus cluster spec") {
+          deriveFixture {
+            runBlocking {
+              export(exportable)
+            }
+          }
+
+          test("has the expected basic properties") {
+            expectThat(kind)
+              .isEqualTo("cluster")
+            expectThat(apiVersion)
+              .isEqualTo(SPINNAKER_TITUS_API_V1)
+            expectThat(spec.locations.regions)
+              .hasSize(2)
+            expectThat(spec.overrides)
+              .hasSize(0)
+          }
+        }
+      }
+
+      context("other handling of default properties in cluster export") {
+        before {
+          coEvery { cloudDriverService.titusActiveServerGroup("us-east-1") } returns activeServerGroupResponseEast
+          coEvery { cloudDriverService.titusActiveServerGroup("us-west-2") } returns activeServerGroupResponseWest
+        }
+
+        test("export omits properties with default values from complex fields") {
+          val exported = runBlocking {
+            export(exportable)
+          }
+          expectThat(exported.spec.defaults.container)
+            .isNotNull()
+
+          expectThat(exported.spec.defaults.iamProfile)
+            .isNotNull()
```

Comment:
If this test is testing that we omit things, should we be checking if things are not null?

Reply:
The mocked clouddriver response contains `iamProfile = moniker.app + "InstanceProfile"` which the export should treat as a default and exclude. The same logic to check if the default iam role is used for ec2 (https://github.com/spinnaker/keel/blob/master/keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt#L492) also applies to titus. Then, this test can be flipped to an `isNull` check as @emjburns noted.

Reply:
@asher you are right, as @emjburns suggested, we will create a different PR to exclude the default fields.